worker_tasks/categorize.py

```diff
@@ -1,11 +1,44 @@
-from collections import Counter
+# categorize.py
+# 
+# Adapted from algorithm by Grineva, et. al. presented at:
+# http://www.slideshare.net/maria.grineva/extracting-key-terms-from-noisy-and-multitheme-documents?related=1
+# 
+# Algorithm Overview
+# 1) Get initial list of possible keyword candidates using ngram frequencies
+# 2) For each keyword candidate, use Wikipedia to extract all related article titles
+#       a) Related articles are called (disambiguation links) on Wikipedia
+# 3) Try to assign one Wikipedia article to each keyword candidate
+#       a) First, assign articles to the keywords with only 1 related article
+#       b) Use these "auto-assigned" articles as the context to narrow the other categories' article
+#       c) With the context from (b), compute relatedness scores to find the best article match
+#           i) relatedness score = [SUM OF SHARED WIKIPEDIA HYPERLINKS] / [TOTAL NUMBER OF WIKIPEDIA HYPERLINKS]
+# 4) With each keyword assigned 1 article, create a graph where each vertex is an article and the edge weight
+#       is the relatedness score between two articles
+# 5) Analyze the graph from (4) to identify keyword clusters
+# 6) Only keep the clusters that have a sufficient number of keywords that contain an original keyword candidate from (1)
+# 8) Return the best keywords
+#       a) For now, we still pick the top two keywords by frequency from kept clusters, but...
+#       b) TODO: WE WILL KEEP ALL KEYWORDS FROM THE KEPT CLUSTERS
+
+
+import logging
+
+from collections import Counter, defaultdict
 from string import punctuation
-from datetime import datetime, timedelta
 
-from sourcemash.models import Item
+import requests
+import urllib
+import json
+from bs4 import BeautifulSoup
 
-BIGRAM_WEIGHT = 1.25
-CAPITALIZED_WORD_WEIGHT = 1.5
+import igraph
+
+logger = logging.getLogger('Sourcemash')
+
+NGRAMS = 3
+MINIMUM_RELATEDNESS_SCORE = 0.1
```

Comment:
What kind of scale do relatedness scores work on? Can you give a range of possible scores (e.g. [0,1], [-inf, inf], [0,10+] meaning that anything past 10 is enormous, etc.)?